docker/llpc-clang-tidy.Dockerfile

```diff
@@ -0,0 +1,62 @@
+#
+# Dockerfile for LLPC Continuous Integration.
+# Sample invocation:
+#    docker build . --file docker/llpc.Dockerfile                                             \
+#                   --build-arg AMDVLK_IMAGE=gcr.io/stadia-open-source/amdvlk:nightly         \
+#                   --build-arg LLPC_REPO_NAME=GPUOpen-Drivers/llpc                           \
+#                   --build-arg LLPC_REPO_REF=<GIT_REF>                                       \
+#                   --build-arg LLPC_REPO_SHA=<GIT_SHA>                                       \
+#                   --build-arg LLPC_BASE_REF=<GIT_REF>                                       \
+#                   --tag llpc-ci/llpc
+#
+# Required arguments:
+# - AMDVLK_IMAGE: Base image name for prebuilt amdvlk
+# - LLPC_REPO_NAME: Name of the llpc repository to clone
+# - LLPC_REPO_REF: ref name to checkout
+# - LLPC_REPO_SHA: SHA of the commit to checkout
+# - LLPC_BASE_REF: ref name for the base of the tested change
+#
+
+# Resume build from the specified image.
+ARG AMDVLK_IMAGE
+FROM "$AMDVLK_IMAGE"
+
+ARG LLPC_REPO_NAME
+ARG LLPC_REPO_REF
+ARG LLPC_REPO_SHA
+ARG LLPC_BASE_REF
+
+# Use bash instead of sh in this docker file.
+SHELL ["/bin/bash", "-c"]
+
+RUN export DEBIAN_FRONTEND=noninteractive && export TZ=America/New_York \
+    && apt-get update \
+    && apt-get install -yqq --no-install-recommends clang-tidy
+
+# Sync the repos. Replace the base LLPC with a freshly checked-out one.
+RUN cat /vulkandriver/build_info.txt \
+    && (cd /vulkandriver && repo sync -c --no-clone-bundle -j$(nproc)) \
+    && git -C /vulkandriver/drivers/llpc remote add origin https://github.com/"$LLPC_REPO_NAME".git \
+    && git -C /vulkandriver/drivers/llpc fetch origin +"$LLPC_REPO_SHA":"$LLPC_REPO_REF" --update-head-ok \
+    && git -C /vulkandriver/drivers/llpc checkout "$LLPC_REPO_SHA"
```

Comment:
This would be the third time we have this code in dockerfiles (already in check-llpc and check-xgl). Could we move this to a short bash/python script?

Reply:
xgl use another path (`/vulkandriver/drivers/xgl`) and other variable names, that would be a bigger change.
The new script gets copied into the amdvlk container, so sharing it with xgl should be possible.